cpp/src/arrow/array/concatenate.cc

```diff
@@ -151,6 +151,10 @@ static Status PutOffsets(const std::shared_ptr<Buffer>& src, Offset first_offset
     return Status::Invalid("offset overflow while concatenating arrays");
   }
 
+  // Concatenate can be called during IPC reads to append delta dictionaries,
+  // on non-validate input.  Be sure to avoid reading out of buffer boundaries.
+  // XXX
```

Comment:
```suggestion
  // Concatenate can be called during IPC reads to append delta dictionaries,
  // on input which has not been validated.  Be sure to avoid reading out of
  // buffer boundaries.
  // XXX
```

Reply:
Ha, I fixed that blunder already :-)